src/silverlabnwb/signature.py

```diff
@@ -250,6 +306,13 @@ def ignored_path(self, path):
                 return True
         return False
 
+    def should_cast_path(self, path, encountered_type):
+        """Should we cast this entity path, and if so, to what type?"""
```

Comment:
Maybe add `:return: either ``None`` if no casting needed or the type to cast to` ?